File: pets/pet-client/test-animation.js
#!/usr/bin/env node

/**
 * 动画测试脚本
 */

const fs = require('fs').promises;
const path = require('path');

// 配置
const config = {
  spritesDir: './sprites',
  manifestFile: './sprites/animation_manifest.json'
};

// 创建目录
async function ensureDir(dirPath) {
  try {
    await fs.mkdir(dirPath, { recursive: true });
  } catch (error) {
    if (error.code !== 'EEXIST') {
      throw error;
    }
  }
}

// 创建简单的占位图片
async function createPlaceholderFrame(stateName, frameNumber) {
  let buffer;
  const hasCanvas = require('canvas');

  if (hasCanvas) {
    // 使用 Canvas 创建占位图
    const { createCanvas } = require('canvas');
    const canvas = createCanvas(180, 180);
    const ctx = canvas.getContext('2d');

    // 清除画布
    ctx.clearRect(0, 0, 180, 180);

    // 根据状态绘制不同的占位图
    ctx.font = '16px Arial';
    ctx.textAlign = 'center';
    ctx.textBaseline = 'middle';

    switch (stateName) {
      case 'idle':
        ctx.fillStyle = '#FF8C42';
        ctx.beginPath();
        ctx.arc(90, 90, 50, 0, Math.PI * 2);
        ctx.fill();
        ctx.fillStyle = '#FFFFFF';
        ctx.fillText('IDLE', 90, 90);
        break;

      case 'idle_long':
        ctx.fillStyle = '#808080';
        ctx.font = 'bold 48px Arial';
        ctx.fillText('Z', 90, 90);
        break;

      case 'working':
        ctx.fillStyle = '#4169E1';
        ctx.fillRect(60, 70, 60, 30);
        ctx.fillStyle = '#FFFFFF';
        ctx.font = '14px Arial';
        ctx.fillText('WORK', 90, 85);
        break;

      case 'thinking':
        ctx.fillStyle = '#9370DB';
        ctx.font = 'bold 48px Arial';
        ctx.fillText('?', 90, 90);
        break;

      case 'success':
        drawStar(ctx, 90, 90, 5, 30, 15);
        ctx.fillStyle = '#FFD700';
        ctx.fill();
        break;

      case 'error':
        ctx.strokeStyle = '#FF0000';
        ctx.lineWidth = 8;
        ctx.beginPath();
        ctx.moveTo(60, 60);
        ctx.lineTo(120, 120);
        ctx.moveTo(120, 60);
        ctx.lineTo(60, 120);
        ctx.stroke();
        break;
    }

    // 添加帧编号
    ctx.fillStyle = '#000000';
    ctx.font = '12px Arial';
    ctx.fillText(`#${frameNumber}`, 15, 15);

    buffer = canvas.toBuffer('image/png');
  } else {
    // 创建简单的文本占位符
    const text = `${stateName.toUpperCase()}\nFrame ${frameNumber}`;
    buffer = Buffer.from(text);
  }

  // 保存文件
  const outputPath = path.join(config.spritesDir, stateName, `frame_${frameNumber.toString().padStart(3, '0')}.png`);
  await fs.writeFile(outputPath, buffer);

  console.log(`  ✅ 创建 ${stateName}/frame_${frameNumber.toString().padStart(3, '0')}.png`);
}

// 绘制星形
function drawStar(ctx, cx, cy, spikes, outerRadius, innerRadius) {
  let rot = Math.PI / 2 * 3;
  let x = cx;
  let y = cy;
  const step = Math.PI / spikes;

  ctx.beginPath();
  ctx.moveTo(cx, cy - outerRadius);

  for (let i = 0; i < spikes; i++) {
    x = cx + Math.cos(rot) * outerRadius;
    y = cy + Math.sin(rot) * outerRadius;
    ctx.lineTo(x, y);
    rot += step;

    x = cx + Math.cos(rot) * innerRadius;
    y = cy + Math.sin(rot) * innerRadius;
    ctx.lineTo(x, y);
    rot += step;
  }

  ctx.lineTo(cx, cy - outerRadius);
  ctx.closePath();
}

// 创建所有占位帧
async function createAllPlaceholderFrames() {
  const states = [
    { name: 'idle', frames: 4 },
    { name: 'idle_long', frames: 4 },
    { name: 'working', frames: 4 },
    { name: 'thinking', frames: 4 },
    { name: 'success', frames: 4 },
    { name: 'error', frames: 4 }
  ];

  console.log('🎨 创建占位帧图片...\n');

  for (const state of states) {
    console.log(`📁 状态: ${state.name} (${state.frames} 帧)`);

    // 确保状态目录存在
    const stateDir = path.join(config.spritesDir, state.name);
    await ensureDir(stateDir);

    // 创建该状态的所有帧
    for (let i = 1; i <= state.frames; i++) {
      await createPlaceholderFrame(state.name, i);
    }
  }
}

// 创建动画清单
async function createAnimationManifest() {
  console.log('\n📋 创建动画清单...');

  const manifest = {
    version: '1.0.0',
    totalFrames: 24,
    states: {
      idle: {
        frameCount: 4,
        duration: 250,
        frames: [
          { path: "idle/frame_001.png", width: 180, height: 180 },
          { path: "idle/frame_002.png", width: 180, height: 180 },
          { path: "idle/frame_003.png", width: 180, height: 180 },
          { path: "idle/frame_004.png", width: 180, height: 180 }
        ]
      },
      idle_long: {
        frameCount: 4,
        duration: 250,
        frames: [
          { path: "idle_long/frame_001.png", width: 180, height: 180 },
          { path: "idle_long/frame_002.png", width: 180, height: 180 },
          { path: "idle_long/frame_003.png", width: 180, height: 180 },
          { path: "idle_long/frame_004.png", width: 180, height: 180 }
        ]
      },
      working: {
        frameCount: 4,
        duration: 200,
        frames: [
          { path: "working/frame_001.png", width: 180, height: 180 },
          { path: "working/frame_002.png", width: 180, height: 180 },
          { path: "working/frame_003.png", width: 180, height: 180 },
          { path: "working/frame_004.png", width: 180, height: 180 }
        ]
      },
      thinking: {
        frameCount: 4,
        duration: 300,
        frames: [
          { path: "thinking/frame_001.png", width: 180, height: 180 },
          { path: "thinking/frame_002.png", width: 180, height: 180 },
          { path: "thinking/frame_003.png", width: 180, height: 180 },
          { path: "thinking/frame_004.png", width: 180, height: 180 }
        ]
      },
      success: {
        frameCount: 4,
        duration: 150,
        frames: [
          { path: "success/frame_001.png", width: 180, height: 180 },
          { path: "success/frame_002.png", width: 180, height: 180 },
          { path: "success/frame_003.png", width: 180, height: 180 },
          { path: "success/frame_004.png", width: 180, height: 180 }
        ]
      },
      error: {
        frameCount: 4,
        duration: 300,
        frames: [
          { path: "error/frame_001.png", width: 180, height: 180 },
          { path: "error/frame_002.png", width: 180, height: 180 },
          { path: "error/frame_003.png", width: 180, height: 180 },
          { path: "error/frame_004.png", width: 180, height: 180 }
        ]
      }
    }
  };

  await fs.writeFile(config.manifestFile, JSON.stringify(manifest, null, 2));
  console.log(`✅ 动画清单已创建: ${config.manifestFile}`);
}

// 检查 Canvas 是否可用
function checkCanvasSupport() {
  try {
    require('canvas');
    return true;
  } catch (error) {
    console.log('⚠️  Canvas 模块未安装，将使用简单的文本占位符');
    return true; // 继续运行，但创建简单的文件
  }
}

// 主函数
async function main() {
  console.log('=== 电子宠物动画测试 ===\n');

  // 检查 Canvas 支持
  if (!checkCanvasSupport()) {
    return;
  }

  try {
    // 创建占位帧
    await createAllPlaceholderFrames();

    // 创建动画清单
    await createAnimationManifest();

    console.log('\n🎉 测试准备完成！');
    console.log('📁 输出目录:', config.spritesDir);
    console.log('📋 清单文件:', config.manifestFile);
    console.log('\n🚀 现在可以启动 electron 客户端测试动画效果');
    console.log('命令: npm start');

    console.log('\n💡 测试步骤:');
    console.log('1. 确保 sprites 目录存在并有图片');
    console.log('2. 启动 electron 客户端 (npm start)');
    console.log('3. 查看控制台日志确认加载了精灵图');
    console.log('4. 通过修改 status.json 切换不同状态');

  } catch (error) {
    console.error('\n❌ 测试失败:', error);
    process.exit(1);
  }
}

// 运行测试
main();

File: pets/pet-client/test_animation.py
#!/usr/bin/env python3
"""
动画测试脚本
用于测试 Minimax 生成的动画帧
"""

import os
import json
import requests
import time
from pathlib import Path


def test_api_connection():
    """测试 Minimax API 连接"""
    print("=== 测试 Minimax API 连接 ===")

    # 测试 API 端点
    url = "https://api.minimax.chat/v1/chat/completions"
    headers = {
        "Authorization": "Bearer sk-cp-Q0OT0DXTp2na1JIBuEzml3b4t9vYXIy8e1llu-yp0J9WKZD1qVnaVSUBXlnLsLo5yKmqph7bhGHm_QysA2kywuZygQSwHfPP9qegB53UOi7kgJA4WtE-8Fs",
        "Content-Type": "application/json"
    }

    data = {
        "model": "abab6.5-chat",
        "messages": [
            {"role": "user", "content": "Hello, can you hear me?"}
        ]
    }

    try:
        response = requests.post(url, json=data, headers=headers, timeout=10)
        if response.status_code == 200:
            print("✅ API 连接成功")
            return True
        else:
            print(f"❌ API 连接失败: {response.status_code}")
            return False
    except Exception as e:
        print(f"❌ API 连接错误: {e}")
        return False


def generate_test_frame():
    """生成一个测试帧"""
    print("\n=== 生成测试帧 ===")

    url = "https://api.minimax.chat/v1/image_generation"
    headers = {
        "Authorization": "Bearer sk-cp-Q0OT0DXTp2na1JIBuEzml3b4t9vYXIy8e1llu-yp0J9WKZD1qVnaVSUBXlnLsLo5yKmqph7bhGHm_QysA2kywuZygQSwHfPP9qegB53UOi7kgJA4WtE-8Fs",
        "Content-Type": "application/json"
    }

    data = {
        "model": "image-01",
        "prompt": "A cute fox-cat sitting idle, fluffy orange fur with white belly, large sparkling eyes, fluffy tail wagging gently, Q-style cartoon, pastel colors, transparent background, looking cute and relaxed",
        "width": 200,
        "height": 200,
        "quality": "standard"
    }

    try:
        response = requests.post(url, json=data, headers=headers, timeout=30)

        if response.status_code == 200:
            result = response.json()
            if "base64_image" in result:
                # 保存测试帧
                import base64
                image_data = base64.b64decode(result["base64_image"])

                test_dir = Path("./sprites/test")
                test_dir.mkdir(parents=True, exist_ok=True)

                with open(test_dir / "test_frame.png", "wb") as f:
                    f.write(image_data)

                print("✅ 测试帧生成成功: sprites/test/test_frame.png")
                return True
            else:
                print(f"❌ API 响应格式错误: {result}")
                return False
        else:
            print(f"❌ API 请求失败: {response.status_code}")
            print(f"响应内容: {response.text}")
            return False
    except Exception as e:
        print(f"❌ 生成测试帧失败: {e}")
        return False


def create_test_animation_manifest():
    """创建测试动画清单"""
    print("\n=== 创建测试动画清单 ===")

    manifest = {
        "version": "1.0.0",
        "totalFrames": 4,
        "states": {
            "idle": {
                "frameCount": 4,
                "duration": 250,
                "frames": [
                    {"path": "test/frame_001.png", "width": 180, "height": 180},
                    {"path": "test/frame_002.png", "width": 180, "height": 180},
                    {"path": "test/frame_003.png", "width": 180, "height": 180},
                    {"path": "test/frame_004.png", "width": 180, "height": 180}
                ]
            }
        }
    }

    # 保存清单
    with open("./sprites/test/animation_manifest.json", "w", encoding="utf-8") as f:
        json.dump(manifest, f, indent=2, ensure_ascii=False)

    print("✅ 测试动画清单已创建: sprites/test/animation_manifest.json")
    return True


def create_placeholder_frames():
    """创建占位帧用于测试"""
    print("\n=== 创建占位帧用于测试 ===")

    try:
        from PIL import Image, ImageDraw

        # 创建占位帧目录
        frames_dir = Path("./sprites/test")
        frames_dir.mkdir(parents=True, exist_ok=True)

        # 为每个状态创建占位帧
        states = ["idle", "idle_long", "working", "thinking", "success", "error"]

        for state in states:
            state_dir = frames_dir / state
            state_dir.mkdir(exist_ok=True)

            # 为每个状态创建 4 帧占位图
            for i in range(1, 5):
                img = Image.new('RGBA', (180, 180), (0, 0, 0, 0))
                draw = ImageDraw.Draw(img)

                # 绘制简单形状表示状态
                if state == "idle":
                    # 坐着的圆形
                    draw.ellipse((30, 40, 150, 160), fill="#FF8C42", outline="#E67E22", width=2)
                    draw.ellipse((70, 60, 110, 80), fill="#FFFFFF")  # 眼白
                    draw.ellipse((75, 65, 85, 75), fill="#000000")  # 眼珠
                elif state == "working":
                    # 站着的方形
                    draw.rectangle((40, 50, 140, 150), fill="#FF8C42", outline="#E67E22", width=2)
                    draw.text((70, 90), "WORK", fill="#FFFFFF", font=None)
                elif state == "success":
                    # 星形
                    draw.polygon([(90, 30), (100, 60), (130, 60), (110, 80), (120, 110), (90, 95), (60, 110), (70, 80), (50, 60), (80, 60)], fill="#FFD700")
                elif state == "error":
                    # X形
                    draw.line((50, 50), (130, 130), fill="#FF0000", width=10)
                    draw.line((130, 50), (50, 130), fill="#FF0000", width=10)
                elif state == "thinking":
                    # 问号
                    draw.text((80, 70), "?", fill="#4169E1", font=None)
                else:  # idle_long
                    # Z形
                    draw.text((70, 80), "Z", fill="#808080", font=None)

                # 添加帧编号
                draw.text((10, 10), f"{i}", fill="#000000", font=None)

                # 保存帧
                img.save(state_dir / f"frame_{i:03d}.png")
                print(f"  创建 {state}/frame_{i:03d}.png")

        print("✅ 所有占位帧创建完成")
        return True

    except ImportError:
        print("❌ PIL 未安装，无法创建占位帧")
        return False


def main():
    print("=== 电子宠物动画测试 ===\n")

    # 1. 测试 API 连接
    if not test_api_connection():
        print("API 连接失败，跳过图片生成")

    # 2. 尝试生成测试帧
    if os.path.exists("./generate_minimax_images.py"):
        generate_test_frame()

    # 3. 创建占位帧（用于测试）
    create_placeholder_frames()

    # 4. 创建测试动画清单
    create_test_animation_manifest()

    print("\n=== 测试完成 ===")
    print("现在可以运行 electron 客户端测试动画效果")
    print("命令: npm start")

    print("\n测试步骤:")
    print("1. 确保 sprites/ 目录存在")
    print("2. 启动 electron 客户端")
    print("3. 检查控制台日志是否加载了精灵图")
    print("4. 切换不同状态看动画效果")


if __name__ == "__main__":
    main()